spec/classes/init_spec.rb

```diff
@@ -672,4 +672,41 @@
       end
     end
   end
+
+  describe 'with disable_monitor set' do
+    let(:facts) { { :osfamily => 'RedHat' } }
+
+    [true,'true',false,'false'].each do |value|
+
+      if !!value == value
+        valuetype = 'boolean'
+      else
+        valuetype = 'string'
+      end
+
+      context "to #{value} as #{valuetype}" do
+        let(:params) { { :disable_monitor => value } }
+
+        it { should compile.with_all_deps }
+        it { should contain_class('ntp') }
+
+        if value == true or value == 'true'
+          it { should contain_file('ntp_conf').with_content(/^disable monitor$/) }
+        else
+          it { should_not contain_file('ntp_conf').with_content(/^disable monitor$/) }
+        end
+
+      end
+    end
+
+    context 'to invalid type' do
+      let(:params) { { :disable_monitor => [ 'i', 'dont', 'like', 'arrays' ] } }
```

Comment:
use `['invalid','type']` as this implies that other types are OK, just not arrays which is not the case.